envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterPermissions.kt

```diff
@@ -47,15 +48,28 @@ class RBACFilterPermissions {
         val matcher = when (matchingType) {
             PathMatchingType.PATH -> StringMatcher.newBuilder().setExact(path).build()
             PathMatchingType.PATH_PREFIX -> StringMatcher.newBuilder().setPrefix(path).build()
+            PathMatchingType.PATH_REGEX -> safeRegexMatcher(path)
         }
         return PathMatcher.newBuilder().setPath(matcher).build()
     }
 }
 
+private fun safeRegexMatcher(regexPattern: String) = StringMatcher.newBuilder()
+    .setSafeRegex(
+        RegexMatcher.newBuilder()
+            .setRegex(regexPattern)
```

Comment:
Should we check the program size? Unfortunately, I can't see the appropriate method in re2j. I suppose we can refine it on a later occasion.

Reply:
We can do it [for sure](https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/matcher/v3/regex.proto.html#envoy-v3-api-field-type-matcher-v3-regexmatcher-googlere2-max-program-size).
But as you said, we can do it later if necessary. If someone will create some monstrous regex, it will have a negative effect only for his/her envoy.

Reply:
Well, we are still passing it in EC. I do not know where the limits are but someone could in theory create a 4GB string and blow up our CP. I would just add a simple `string.length` check (e.g. 4096 chars max, make it configurable of course).

Reply:
But it is also true for every other field in proxy settings. Why treat this field differently?

Reply:
I think you're right, but in this field I saw a greater potential for "creativity". Let's just YAGNI it and react when needed.